File: src/main/java/hello/Builder/UserBuilder.java
package hello.Builder;

import hello.User;

public class UserBuilder {

    User user;

    public UserBuilder() {
        this.user = new User();
    }

    public User build() {
        return this.user;
    }

    public UserBuilder id(Integer id) {
        this.user.setId(id);
        return this;
    }

    public UserBuilder name(String name) {
        this.user.setName(name);
        return this;
    }

    public UserBuilder age(Integer age) {
        this.user.setAge(age);
        return this;
    }
}

File: src/main/java/hello/Builder/UserBuilder.kt
package hello.Builder

import hello.User

class UserBuilder {
    var user: User
    fun build(): User {
        return user
    }

    fun id(id: Int?): UserBuilder {
        user.id = id
        return this
    }

    fun name(name: String?): UserBuilder {
        user.name = name
        return this
    }

    fun age(age: Int?): UserBuilder {
        user.age = age
        return this
    }

    init {
        user = User()
    }
}